lhotse/workflows/whisper.py

```diff
@@ -64,12 +79,44 @@ def annotate_with_whisper(
             supervisions = _postprocess_timestamps(supervisions)
             cut.supervisions = list(
                 trim_supervisions_to_recordings(
-                    recordings=recordings, supervisions=supervisions, verbose=False
+                    recordings=recording, supervisions=supervisions, verbose=False
                 )
             )
         yield cut
 
 
+def _annotate_cuts(cuts: CutSet, language: str, model_name: str, device: str):
+    """
+    Helper function that annotates a CutSet with Whisper.
+    """
+    import whisper
+
+    model = whisper.load_model(model_name, device=device)
+
+    for cut in cuts:
+        if cut.num_channels > 1:
+            logging.warning(
+                f"Skipping cut '{cut.id}'. It has {cut.num_channels} channels, "
+                f"but we currently only support mono input."
+            )
+            continue
+        audio = torch.from_numpy(cut.resample(16000).load_audio()).squeeze(0)
+        result = whisper.transcribe(model=model, audio=audio, language=language)
+        supervisions = [
+            SupervisionSegment(
+                id=f"{cut.id}-{segment['id']:06d}",
+                recording_id=cut.recording_id,
+                start=round(segment["start"], ndigits=8),
+                duration=round(segment["end"], ndigits=8),
```

Comment:
I think with this single change we can prevent timestamps going out of cut's duration; can you also include `_postprocess_timestamps` in the cut-based workflow?

```suggestion
                duration=max(cut.duration, round(segment["end"], ndigits=8)),
```

Reply:
Done!

Reply:
Because in my own testing I discovered that their "end" is actually a "duration" 😂 but if you could triple-check that I got it right, maybe using some longer recording, that'd be great.

Reply:
So I ran the model on one of the AMI headset recordings (~5000s) and it seems like the "end" actually shows the end of the segment, not the duration. Here is the JSON containing the `results["segments"]`: https://drive.google.com/file/d/169igkcDY2SmMs5k3hOhHip89T4MQDnKs/view?usp=sharing